dwave/cloud/client.py

```diff
@@ -1162,31 +1233,22 @@ def filter_ready(item):
                 logger.debug("Submitting %d problems", len(ready_problems))
                 try:
                     body = '[' + ','.join(mess.body.result() for mess in ready_problems) + ']'
-                    try:
-                        response = session.post('problems/', body)
-                        localtime_of_response = epochnow()
-                    except requests.exceptions.Timeout:
-                        raise RequestTimeout
-
-                    if response.status_code == 401:
-                        raise SolverAuthenticationError()
-                    response.raise_for_status()
-
-                    message = response.json()
+                    logger.debug('Size of POST body = %d', len(body))
+                    message = Client._sapi_request(session.post, 'problems/', body)
                     logger.debug("Finished submitting %d problems", len(ready_problems))
-                except BaseException as exception:
+
+                except Exception as exc:
                     logger.debug("Submit failed for %d problems", len(ready_problems))
-                    if not isinstance(exception, SolverAuthenticationError):
-                        exception = IOError(exception)
+                    if not isinstance(exc, SolverAuthenticationError):
+                        exc = IOError(exc)
 
                     for mess in ready_problems:
-                        mess.future._set_exception(exception)
+                        mess.future._set_exception(exc)
```

Comment:
You standardly use `msg`, so I parsed this as there being a mess in ready_problems and wondered if I'd missed a definition of what you consider messy.

Reply:
mess was inherited from the micro client